rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/importexport/Sw360ImportExportService.java

```diff
@@ -0,0 +1,253 @@
+            for (Component component : componentDetailedSummaryForExport) {
+                printComponentAttachments(component, csvRows);
+                printReleasesAttachments(component, csvRows);
+            }
+        }
+        ByteArrayInputStream byteArrayInputStream = CSVExport.createCSV(ComponentAttachmentCSVRecord.getCSVHeaderIterable(), csvRows);
+        String filename = String.format(fileConstant, SW360Utils.getCreatedOn());
+        response.setHeader("Content-Disposition", String.format("Attachment; filename=\"%s\"", filename));
+        FileCopyUtils.copy(byteArrayInputStream, response.getOutputStream());
+    }
+
+    private void printReleasesAttachments(Component component, List<Iterable<String>> csvRows) {
+        final List<Release> releases = component.getReleases();
+        if (releases != null && !releases.isEmpty()) {
+            for (Release release : releases) {
+                printReleaseAttachments(release, csvRows);
+            }
+        }
+    }
+
+    private void printReleaseAttachments(Release release, List<Iterable<String>> csvRows) {
+        final Set<Attachment> attachments = release.getAttachments();
+        printAttachments(attachments, csvRows, builder -> builder.fill(release));
+
+    }
+
+    private void printComponentAttachments(Component component, List<Iterable<String>> csvRows) {
+        final Set<Attachment> attachments = component.getAttachments();
+        printAttachments(attachments, csvRows, builder -> builder.fill(component));
+
+    }
+
+    private void printAttachments(Set<Attachment> attachments, List<Iterable<String>> csvRows,  Consumer<ComponentAttachmentCSVRecordBuilder> containingObjectPrinter) {
+        if (attachments != null && !attachments.isEmpty()) {
+            for (Attachment attachment : attachments) {
+                final ComponentAttachmentCSVRecordBuilder componentAttachmentCSVRecordBuilder = ComponentAttachmentCSVRecord
+                        .builder();
+                containingObjectPrinter.accept(componentAttachmentCSVRecordBuilder);
+                componentAttachmentCSVRecordBuilder.fill(attachment);
+                csvRows.add(componentAttachmentCSVRecordBuilder.build().getCSVIterable());
+            }
+        }
+    }
+
+    private List<Component> getComponentDetailedSummaryForExport() {
+        final ComponentService.Iface componentClient  = getThriftComponentClient();
+
+        final List<Component> componentDetailedSummaryForExport;
+        try {
+            componentDetailedSummaryForExport = componentClient.getComponentDetailedSummaryForExport();
+        } catch (TException e) {
+            log.error("Problem fetching components", e);
+            return null;
+        }
+
+        return componentDetailedSummaryForExport;
+    }
+
+    private Iface getThriftComponentClient() {
+        ComponentService.Iface componentClient = new ThriftClients().makeComponentClient();
```

Comment:
Immediately return this expression instead of assigning it to the temporary variable "**componentClient**".